### PR TITLE
Update README.md

### LINKED ISSUE #8: Update ReadMe
## Feature Description

Update ReadMe file to include full details about client. Also add EXAMPLES.md for examples about each available function.

---

## Use Case

N/A

---

## Proposed Solution

N/A

---

## Additional Context

N/A

---

## Pseudo Code

```go
// Provide a code snippet or example that demonstrates how this feature might be used

### PR DESCRIPTION
**Pull request by that-ar-guy**
fixes: #8 

## Summary by CodeRabbit

- **Documentation**
	- Updated the README to reflect rebranding from "Go API Client" to "Go API Weather."
	- Added new sections: "Features," "Installation," "Usage," "Configuration," "Supported Functions," "Contributing," "License," and "Acknowledgments."
	- Improved clarity and usability of installation and usage instructions, including example code updates and error handling enhancements.
	- Introduced a new `EXAMPLES.md` file with practical examples for using the Go-API Client, covering current weather retrieval, 5-day forecasts, and error handling.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,33 +1,68 @@
...
+
+```
+client := weather.NewClient("YOUR_API_KEY")
+```
+
+## Supported Functions
+- `GetCurrentWeather(location string)`: Fetch current weather for a given location.
+- `GetForecast(location string, days int)`: Retrieve weather forecast for a specified number of days.
+## Examples
+Detailed usage examples for each function are provided in the `EXAMPLES.md` file.
```
[the remark this answers is not in the document]
Comment (that-ar-guy) [reply README.md]: `EXAMPLES.md` created

File: README.md [hunk 2]
```diff
@@ -1,33 +1,68 @@
...
-### API Docs:
-[WeatherAPI Documentation](https://www.weatherapi.com/docs/)
+Go-Weather is a simple and efficient Go library for fetching weather information from various sources. This library allows developers to seamlessly integrate weather data into their applications.
 
-### Usage:
-### Download:
+## Features
+- Fetch current weather for any location.
+- Retrieve forecast data for up to 7 days.
+- Support for multiple weather APIs.
```
[the remark this answers is not in the document]
Comment (that-ar-guy) [reply README.md]: resolved

File: EXAMPLES.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
+
+
+
+# Examples for Go-API Client
```
Comment (kashifkhan0771) [inline EXAMPLES.md:1]: Examples for Go WeatherAPI Client

File: README.md [hunk 4]
```diff
@@ -1,33 +1,68 @@
...
-### Usage:
-### Download:
+## Features
+- Fetch current weather for any location.
+- Retrieve forecast data for up to 7 days.
+- Supports WeatherAPI integration for reliable weather data.
+- Lightweight and easy-to-use interface.
+
+## Installation
+To use Go-Weather in your project, ensure you have Go installed and run:
```
Comment (kashifkhan0771) [inline README.md]: Add go version details as well.

File: README.md [hunk 5]
```diff
@@ -1,33 +1,68 @@
-# Go API Client
+# Go API Weather
 
-### API Docs:
-[WeatherAPI Documentation](https://www.weatherapi.com/docs/)
+Go-Weather is a simple and efficient Go library for fetching weather information from various sources. This library allows developers to seamlessly integrate weather data into their applications.
```
Comment (kashifkhan0771) [inline README.md]: > weather information from various sources
weather information from https://www.weatherapi.com

File: README.md [hunk 6]
```diff
@@ -1,33 +1,68 @@
...
 
-### Usage:
-### Download:
+## Features
+- Fetch current weather for any location.
+- Retrieve forecast data for up to 7 days.
+- Supports WeatherAPI integration for reliable weather data.
+- Lightweight and easy-to-use interface.
```
Comment (kashifkhan0771) [inline README.md]: > Lightweight and easy-to-use interface.
Lightweight

**Review by kashifkhan0771 — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (that-ar-guy) [reply README.md]: done

File: README.md (hunk 5 above)
Comment (that-ar-guy) [reply README.md]: done

File: README.md (hunk 6 above)
Comment (that-ar-guy) [reply README.md]: done

File: EXAMPLES.md (hunk 3 above)
Comment (that-ar-guy) [reply EXAMPLES.md:1]: done ?

**Review by kashifkhan0771 — APPROVED** [review]

**Comment by kashifkhan0771** [comment]
Thank you @that-ar-guy for the contribution ❤️

**Comment by that-ar-guy** [comment]
Thank you for reviewing and merging my pull request. I apologize if my work seemed less polished, as I am still a beginner. I would greatly appreciate any feedback you could provide to help me improve.

Thank you again for the opportunity to contribute.

Best regards,
[AR](https://www.github.com/that-ar-guy)

**Comment by kashifkhan0771** [comment]
No worries @that-ar-guy your work was up-to mark. Slowly we all learn. I merged the PR. If you have any improvements in your mind at anytime, feel free to open an issue and create a PR.

Happy Hacking!

**Pull request merged**